### ISSUE TITLE
Specifying types within a types.MappingProxyType?

### ISSUE THREAD

**Issue by madig** [python/mypy#5220]
I have this code:
```Python
import types

class A:
    a: Dict[str, Any]

x=A()
x.a = types.MappingProxyType({"aaa": 1})
```
I want to store a read-only view of a dict in that class. Mypy complains that the types are incompatible and I should declare `a` to be:
```Python
class A:
    a: types.MappingProxyType[str, Any]
```
That will however generate `"TypeError: 'type' object is not subscriptable"`. I can drop the `[str, Any]` but I'd like to keep it. Is that possible?

Python 3.6.5 On Win10 x64, mypy 0.610 (issue also on master), args: `--ignore-missing-imports --follow-imports=silent`.

**Comment by JelleZijlstra** [comment]
You probably want something like `Mapping[str, Any]` or `MutableMapping[str, Any]` (both from `typing`).

**Comment by madig** [comment]
Wonderful, that did the trick! Maybe mypy should detect `types.MappingProxyType`and suggest `typing.Mapping` instead.
